### PR TITLE
Enable api

### PR DESCRIPTION
**Pull request by NoumbissiValere**

### CODE REVIEW

File: template_library/views.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    adapter_class = FacebookOAuth2Adapter
+
+
+class GoogleLogin(SocialLoginView):
+    adapter_class = GoogleOAuth2Adapter
+
+
+class CustomConfirmEmailView(ConfirmEmailView):
+    """
+    Override default register view to
```
Comment (nemesifier) [inline template_library/views.py]: the docstring looks incorrect, it mentions the register view but the view is inheriting `ConfirmEmailView`, also, the name `CustomConfirmEmailView` is not great, I'd prefer to have:

```python
from allauth.account.views import BaseConfirmEmailView


class ConfirmEmailView(BaseConfirmEmailView):
    ...
```

File: template_library/views.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    redirect users to create organizations
+    after signup
+    """
+
+    def post(self, *args, **kwargs):
+        redirect_url = 'list_orgs'
```
Comment (nemesifier) [inline template_library/views.py]: this doesn't look right, it looks like are we redirecting to a URL that is just `list_orgs`, can you expand on this?

**Review by nemesifier — COMMENTED** [review]
where are the following endpoints?

- password forgotten
- change password
- login
- logout

File: template_library/views.py (hunk 2 above)
Comment (NoumbissiValere) [reply template_library/views.py]: the url points to a list/create view. the GET request will return an empty list, will use this information to load the form in the frontend. this form will be POST to the list_orgs. to create the orgs. I think the name should have been `list_create_orgs`

**Comment by NoumbissiValere** [comment]
> where are the following endpoints?
> 
> * password forgotten
> * change password
> * login
> * logout

Those endpoints have been implemented in the `rest-auth`. so from [here](https://django-rest-auth.readthedocs.io/en/latest/api_endpoints.html), i can call them with the respective methods and params

**Comment by nemesifier** [comment]
> Those endpoints have been implemented in the rest-auth. so from here, i can call them with the respective methods and params

From the URL routes I can't see these endpoints in this project, where are these being imported?

File: template_library/views.py (hunk 2 above)
Comment (nemesifier) [reply template_library/views.py]: I still don't get it, if you do redirect('list_orgs'), won't the user be redirected to `http://list_orgs`?

Comment (NoumbissiValere) [reply template_library/views.py]: `list_orgs` is a view name found in `openwisp_users`  i am passing a view name to redirect and not a url path

**Comment by NoumbissiValere** [comment]
> where are the following endpoints?
> 
> * password forgotten
> * change password
> * login
> * logout

Those endpoints have been implemented in the `rest-auth`. so from [here](https://django-rest-auth.readthedocs.io/en/latest/api_endpoints.html), i can call them with the respective methods and params

> > Those endpoints have been implemented in the rest-auth. so from here, i can call them with the respective methods and params
> 
> From the URL routes I can't see these endpoints in this project, where are these being imported?

Those are included in this line:
```url(r'^rest-auth/', include('rest_auth.urls'))```
in `template_library.urls`

**Comment by NoumbissiValere** [comment]
@nemesisdesign i have done the requested changes please can you review? Thanks

**Review by nemesifier — COMMENTED** [review]
please fix the conflicts and update the PR

**Comment by NoumbissiValere** [comment]
@nemesisdesign please this is ready for review

File: template_library/generics.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        options = {
+            'description': request.POST.get('des', None),
+            'name': request.POST.get('name', None),
+            'slug': request.POST.get('slug', None),
+            'email': request.POST.get('email', None),
+            'url': request.POST.get('org-url', None)
+        }
+        org = self.org_model(**options)
+        try:
+            org.full_clean()
```
Comment (nemesifier) [inline template_library/generics.py]: this is wrong, it needs to be done with a serializer, please study the DRF docs and tutorials to find out how to do it properly

**Review by nemesifier — CHANGES_REQUESTED** [review]
Good progress, see my comments

**Review by nemesifier — APPROVED** [review]
go ahead

**Pull request merged**